src/main/java/org/jenkinsci/plugins/casc/plugins/PluginManagerConfigurator.java

```diff
@@ -144,7 +142,11 @@ public PluginManager configure(CNode config) throws ConfiguratorException {
             install:
             while (!plugins.isEmpty()) {
                 PluginToInstall p = plugins.remove();
-                if (installed.contains(p.shortname)) continue;
+                logger.info("Preparing to install "+ p.shortname);
+                if (installed.contains(p.shortname)) {
```

Comment:
In such case installed pligins won't be updated if needed

Reply:
@oleg-nenashev I just tried, it does not seem to break upgrading. I tried to update to git 3.9.1 from the example above and it worked as i expected.

Besides, i did not change any logic in that part of the code, so this flaw is already on master?

Reply:
I tried doing it using the `reload` button in JCasC

Reply:
@oleg-nenashev I see your point. It does not check version of the plugin installed, so if a plugin requires a never version of a plugin, it will not upgrade. I'll see if i can fix it.

Reply:
@oleg-nenashev `installed` in this context is the set of plugin we just installed, not the one already installed on this master

Reply:
@oleg-nenashev A point to make is that this code is no worse than what already is on master. Except that i can actually configure a Jenkins instance with plugins: section with my changes.